gcc/config/mingw/winnt.cc

```diff
@@ -807,6 +809,13 @@ mingw_pe_file_end (void)
 	  if (!startswith (name, "refptr."))
 	    continue;
 	  name += 7;
+
+	  if (q->is_weak)
+	    {
+	      fprintf (asm_out_file, "\t.weak\t%s\n", name);
```

Comment:
Why `ASM_WEAKEN_LABEL` could not be used instead? As far as I know the compiler was producing `.weak` directive for weak symbols before this change. What was wrong with that directives?

Reply:
ASM_WEAKEN_LABEL is still used by compiler for some cases, however for cases when weak symbols are declared in the C\C++ code it is not happening anymore. The reason is weak symbols are replaced fully by references in aarch64 to produce proper relocations, which is different in ix86.  In ix86 case symbols can be tracked and declared as weak.
For aarch64 we need to track them separately and declare with stubs.
The implementation has been done in mingw code style, however it might be a good idea to change the style and use  ASM_WEAKEN_LABEL at that place.